libc-bottom-half/cloudlibc/src/common/net.h

```diff
@@ -64,12 +65,18 @@ static inline int wasi_to_sockaddr(const struct __wasi_addr_port_t *restrict pee
     } else if (peer_addr->tag == __WASI_ADDRESS_FAMILY_INET6) {
       struct sockaddr_in6 addr6;
       addr6.sin6_family = AF_INET6;
-      addr6.sin6_flowinfo = 0;
-      addr6.sin6_scope_id = 0;
+      addr6.sin6_flowinfo = need_revert?htonl(peer_addr->u.inet6.addr.flow_info):peer_addr->u.inet6.addr.flow_info;
+      addr6.sin6_scope_id = need_revert?htonl(peer_addr->u.inet6.addr.scope_id):peer_addr->u.inet6.addr.scope_id;;
```

Comment:
You are sure the `need_revert` test also apply to those new field?

Reply:
I'm 99% sure, but let me make 100% sure all the same.